darwin/cli.py

```diff
@@ -22,61 +36,73 @@ def main():
-        print(path)
-
-    # Print the url of a remote project
-    elif args.command == "url":
-        url = f.url(args.project_name)
-        print(url)
-
-    # Download dataset
-    elif args.command == "pull":
-        project_name = args.project_name
-        local_dataset = f.pull_project(project_name)
-        print(f"Project {project_name} downloaded at {local_dataset.local_path}. ")
-
-    # List existing projects (remotely)
-    elif args.command == "remote":
-        f.remote()
-
-    # Remove a project (remotely)
-    elif args.command == "remove":
-        project_name = args.project_name
-        f.remove_remote_project(project_name)
-
-    # Upload new data to a project (remotely)
-    elif args.command == "push":
-        f.upload_data(args.project_name, args.files, args.exclude, args.fps)
-
+        else:
+            f.list_teams()
     # Version
     elif args.command == "version":
-        print("0.0.1")
+        print("0.3")
+
+    elif args.command == "dataset":
+        if args.action == "remote":
+            f.list_remote_datasets(args.all, args.team)
+        elif args.action == "local":
+            f.local()
+        elif args.action == "create":
+            f.create_dataset(args.dataset_name, args.team)
+        elif args.action == "path":
+            path = f.path(args.dataset)
+            if path:
+                print(path)
+            else:
+                print("The dataset has not been downloaded")
+        # Print the url of a remote project
+        elif args.action == "url":
+            f.url(args.dataset)
+        elif args.action == "push":
+            f.upload_data(args.dataset, args.files, args.exclude, args.fps)
+        # Remove a project (remotely)
+        elif args.action == "remove":
+            f.remove_remote_dataset(args.dataset)
+        elif args.action == "report":
+            f.dataset_report(args.dataset, args.granularity or "day")
+        elif args.action == "releases":
+            f.dataset_list_releases(args.dataset)
+        elif args.action == "pull":
+            f.pull_dataset(args.dataset)
+        elif args.action == "help" or args.action == None:
+            dataset_parser = [
+                action.choices["dataset"]
+                for action in parser._actions
+                if isinstance(action, argparse._SubParsersAction) and "dataset" in action.choices
+            ][0]
+
+            subparsers_actions = [
+                action
+                for action in dataset_parser._actions
+                if isinstance(action, argparse._SubParsersAction)
+            ]
+
+            print(dataset_parser.description)
+            print("\nCommands:")
+            for subparsers_action in subparsers_actions:
+                # get all subparsers and print help
+                for choice in sorted(subparsers_action._choices_actions, key=lambda x: x.dest):
+                    print("    {:<19} {}".format(choice.dest, choice.help))
```

Comment:
Wouldn't still make sense to move this piece of code somewhere else?

Reply:
done